### PR TITLE
update hypershift e2e

### PR DESCRIPTION
**Pull request by LiangquanLi930**
1. `idp-htpasswd` not support hypershift guest cluster, so remove it
2. Sort the job in alphanumeric order

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
@heliubj18 @kuiwang02 @liangxia PTAL this PR, thanks

**Comment by heliubj18** [comment]
LGTM

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml [hunk 1]
```diff
@@ -342,9 +342,21 @@ tests:
       TEST_TIMEOUT: "30"
     test:
     - ref: cucushift-helper-service-build-info
-    - ref: idp-htpasswd
     - ref: openshift-extended-test
     workflow: cucushift-installer-rehearse-aws-ipi-ovn-hypershift-guest
+- as: e2e-aws-ipi-ovn-hypershift-install-p2
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:347]: Looks like there are sort issue ?

Comment (LiangquanLi930) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:347]: You mean to put between `e2e-aws-ipi-ovn-fips-p3-destructive` and `e2e-aws-ipi-ovn-ingress-nlb-p1` right?

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:347]: Yes, as `e2e-aws-ipi-ovn-fips-xxx` sorts before `e2e-aws-ipi-ovn-hypershift-xxx` ?

Comment (LiangquanLi930) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:370]: @liangxia could you help to review again? thanks

**Pull request merged**